src/pages/Home/components/SearchBarModal.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useRef, useState } from 'react';
+import { useQuery } from '@tanstack/react-query';
+import clsx from 'clsx';
+import { useNavigate } from 'react-router-dom';
+import SearchBox from './SearchBox';
+import { getRelateSearch } from '@/api/shop';
+import { RelatedSearchResponse } from '@/api/shop/entity';
+import Header from '@/components/Layout/Header';
+import Portal from '@/components/Portal';
+import RelateSearchItem from '@/pages/Home/components/RelateSearchItem';
+import useDebouncedCallback from '@/util/hooks/useDebounce';
+import useEscapeKeyDown from '@/util/hooks/useEscapeKeyDown';
+import useHandleOutside from '@/util/hooks/useHandleOutside';
+import useScrollLock from '@/util/hooks/useScrollLock';
+
+interface SearchBarModalProps {
+  onClose: () => void;
+}
+
+function useModalClose(onClose: () => void) {
+  const backgroundRef = useRef<HTMLDivElement>(null);
+  const containerRef = useRef<HTMLDialogElement>(null);
+  useScrollLock(true);
+  useEscapeKeyDown(onClose);
+
+  useHandleOutside<HTMLDialogElement, HTMLDivElement>({
+    containerRef,
+    backgroundRef,
+    onOutsideClick: (e) => {
+      e.preventDefault?.();
+      onClose();
+    },
+  });
+  return { backgroundRef, containerRef };
+}
+
+function useSearch(delay = 200) {
+  const [keyword, setKeyword] = useState('');
+
+  const setDebounced = useDebouncedCallback((value: string) => {
+    setKeyword(value.trim());
+  }, delay);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setDebounced(e.target.value);
+  };
+
+  return { keyword, onChange };
+}
+
+export default function SearchBarModal({ onClose }: SearchBarModalProps) {
+  const { backgroundRef, containerRef } = useModalClose(onClose);
+  const navigate = useNavigate();
+
+  const { keyword, onChange } = useSearch(200);
+
+  const { data } = useQuery<RelatedSearchResponse>({
+    queryKey: ['relateSearch', keyword],
+    queryFn: () => getRelateSearch({ keyword }),
+    enabled: keyword.length > 0,
+  });
+
+  const relateSearchItems = data;
```

Comment:
이름을 바꾸어서 사용하고 싶다면 아래와 같이 하면 될 것 같아요!
그리고 어법상 `relatedSearchItems`가 맞는것 같아요.


```suggestion
  const { data: relatedSearchItems } = useQuery<RelatedSearchResponse>({
    queryKey: ['relateSearch', keyword],
    queryFn: () => getRelateSearch({ keyword }),
    enabled: keyword.length > 0,
  });
```